### ISSUE TITLE
No error event for client-timeout

### ISSUE THREAD

**Issue by TamingOfTheShru** [wankdanker/node-sftpjs#11]
Hi, 
I've been trying to connect to a sftp server for which the module works great. 
But when there is a timeout while connecting, no event is emitted for such error. I need to handle these kinds of errors as per my system requirements. 
If we modify following code in sftp.js:

```
self.conn.on('error', function (err) {
		//ignore errors if the connection has been closed
		//this is to prevent a common ECONNRESET error that
		//occurs.
		if (self.closed) {
			return;
		}
		self.emit('error', err);
	});

```
Upon commenting out the "if block", the issue's fixed. Can you update the code base accordingly?

**Comment by wankdanker** [comment]
I see what you are saying. There was a reason the if block was put there, but I think I need to reconsider that logic. It was done in this commit: adc3c39a73912bb1a9430242c1ea53d1d9205fea

I think what happens is when we intentionally close a connection, we end up getting errors propagating from SSH2. This was intended to fix that. I think that maybe `self.closed` should just be set to `false` when a connection attempt has been started.

**Comment by wankdanker** [comment]
Just published sftpjs@1.3.6 to npm. Give that a try and re-open this issue if it is not fixed.

Thanks!

**Comment by TamingOfTheShru** [comment]
Perfect! Thank you!
